ecosystem/typescript/sdk/src/transaction_builder/builder_utils.ts

```diff
@@ -161,7 +161,7 @@ function serializeStruct(argVal: any, argType: TypeTag, serializer: Serializer,
 function serializeOption(argVal: any, argType: TypeTag, serializer: Serializer, depth: number) {
   // For option, we determine if it's empty or not empty first
   // empty option is nothing, we specifically check for undefined to prevent fuzzy matching
-  if (argVal === undefined) {
+  if (argVal === undefined || argVal === null) {
```

Comment:
might be easier
```suggestion
  if (!argVal) {
```

Reply:
We have be very careful about things like this, and probably should audit them.

This argval could be `""` or `0`, which are both falsey.

Reply:
+1 for being explicit